File: src/CalibrationResult.cs
// CalibrationResult.cs
// MetrologyCalibration - ModMind Data Standard
using System;
using System.Text.Json.Serialization;

public struct MatrixElement {
    [JsonPropertyName("m")]
    public float Value { get; set; }
}

public struct CalibrationResult {
    [JsonPropertyName("timestamp")]
    public DateTime Timestamp { get; set; }
    [JsonPropertyName("rmse")]
    public double Rmse { get; set; }
    [JsonPropertyName("transform_elements")]
    public MatrixElement[] TransformElements { get; set; }
    [JsonPropertyName("source_platform")]
    public string SourcePlatform { get; set; }
}

File: src/CalibrationResult.swift
// CalibrationResult.swift
// MetrologyCalibration - ModMind Data Standard
import Foundation

struct MatrixElement: Codable {
    let m: Float
}

struct CalibrationResult: Codable {
    let timestamp: Date
    let rmse: Double
    let transform_elements: [MatrixElement]
    let source_platform: String
}
